fuse_core/include/fuse_core/throttled_callback.h

```diff
@@ -200,12 +214,37 @@ class ThrottledCallback
 };
 
 /**
- * @brief Throttled callback for ROS messages
+ * @brief Throttled callback for ROS messages, option to throttle using header timestamps with `message_stamp_callback`
  *
  * @tparam M The ROS message type, which should have the M::ConstPtr nested type
  */
 template <class M>
-using ThrottledMessageCallback = ThrottledCallback<std::function<void(const typename M::ConstPtr&)>>;
+class ThrottledMessageCallback : public ThrottledCallback<std::function<void(const typename M::ConstPtr&)>>
+{
+public:
+  using Callback = std::function<void(const typename M::ConstPtr&)>;
+
+  ThrottledMessageCallback(Callback&& keep_callback = nullptr,  // NOLINT(whitespace/operators)
+                           Callback&& drop_callback = nullptr,  // NOLINT(whitespace/operators)
+                           const ros::Duration& throttle_period = ros::Duration(0.0), const bool use_wall_time = false)
+    : ThrottledCallback<Callback>(std::move(keep_callback), std::move(drop_callback), throttle_period, use_wall_time)
+    {
+    }
+
+  /**
+   * @brief Callback that uses message header timestamps to throttle the calls to the keep callback provided.
+   * When dropped because of throttling, the drop callback is called instead.
+   * @details The first message after the most recent whole-number multiple of the throttle_period_ is published
+   * and all others are dropped.
+   *
+   * @param[in] msg The input arguments
+   */
+  void message_stamp_callback(const typename M::ConstPtr& msg)
```

Comment:
This is a good idea, but where is this method called?

I think we should also rename it to
```suggestion
  void messageStampCallback(const typename M::ConstPtr& msg)
```

And we should also check the message type `M` has `header.stamp` - you can use https://docs.ros.org/en/indigo/api/roscpp_traits/html/message__traits_8h_source.html#l00106 to `enable_if` this method or the entire subclass only for messages with header, which will have a stamp

Reply:
It's not called yet! I wanted to check the existing test cases with the new default logic and I was fighting to get the tests running. Now that they are running I will add a test to this module then look at adopting in the sensor models.

For adoption in the sensor models, we should probably add a config option to select which callback to use then update our configs to use the new one on our side after that goes in. I'm guessing we will also have to update some expected test results. Does that sound like a reasonable approach?